src/Common/src/System/Collections/Generic/LowLevelDictionary.IDictionary.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            var result = new LowLevelDictionary<TKey, TValue>(_numEntries);
+            for (int bucket = 0; bucket < _buckets.Length; bucket++)
+            {
+                for (Entry entry = _buckets[bucket]; entry != null; entry = entry._next)
+                {
+                    result.Add(entry._key, entry._value);
+                }
+            }
+            return result;
+        }
+
+        bool ICollection.IsSynchronized => false;
+
+        object ICollection.SyncRoot => this;
+
+        void IDictionary.Add(object key, object value) => Add((TKey)key, (TValue)value);
+
+        bool IDictionary.Contains(object key) => Find((TKey)key) != null;
+
+        IDictionaryEnumerator IDictionary.GetEnumerator() => new DictionaryEnumerator(this);
+        IEnumerator IEnumerable.GetEnumerator() => new DictionaryEnumerator(this);
+
+        void IDictionary.Remove(object key) { Remove((TKey)key); }
+
+        void IDictionary.Clear() => Clear();
+
+        void ICollection.CopyTo(Array array, int index)
+        {
+            int dst = 0;
+            for (int bucket = 0; bucket < _buckets.Length; bucket++)
+            {
+                for (Entry entry = _buckets[bucket]; entry != null; entry = entry._next)
+                {
+                    array.SetValue(new DictionaryEntry(entry._key, entry._value), dst++);
+                }
+            }
+        }
+
+        private sealed class DictionaryEnumerator : IDictionaryEnumerator
+        {
+            private readonly DictionaryEntry[] _entries;
+            private int _pos = -1;
+
+            internal DictionaryEnumerator(LowLevelDictionary<TKey, TValue> dict)
+            {
+                var entries = new DictionaryEntry[dict._numEntries];
+                int dst = 0;
+                for (int bucket = 0; bucket < dict._buckets.Length; bucket++)
+                {
+                    for (Entry entry = dict._buckets[bucket]; entry != null; entry = entry._next)
+                    {
+                        entries[dst++] = new DictionaryEntry(entry._key, entry._value);
+                    }
+                }
+                _entries = entries;
+            }
+
+            public object Current => Entry;
+            public object Key => Entry.Key;
+            public object Value => Entry.Value;
+
+            public DictionaryEntry Entry
+            {
+                get
+                {
+                    if (_pos < 0 || _pos >= _entries.Length)
+                    {
+                        throw new InvalidOperationException();
+                    }
+                    return _entries[_pos];
+                }
+            }
+
+            public bool MoveNext()
+            {
+                if (_pos < _entries.Length)
+                {
+                    _pos++;
+                }
+                return _pos < _entries.Length;
+            }
+
+            public void Reset() { throw new NotSupportedException(); }
```

Comment:
> Would it be easier to just switch to regular Dictionary at this point? I don't think there's a layering issue.

I'd be happy to and would like to, but I can't until @KrzysztofCwalina's work lands to combine all of the relevant assemblies.  As it currently stands, System.Runtime.Extensions doesn't have access to it.

But that's also why I haven't done a lot of work on the implementation here around argument validation and the like, because the whole LowLevelDictionary thing should completely go away.